olc_art.hpp

```diff
@@ -264,6 +759,72 @@ class olc_db final {
   friend struct detail::olc_impl_helpers;
 };
 
+///
+/// ART iterator implementation.
+///
+
+inline std::optional<const unodb::key> olc_db::iterator::get_key() noexcept {
+  // Note: If the iterator is on a leaf, we return the key for that
+  // leaf regardless of whether the leaf has been deleted.  This is
+  // part of the design semantics for the OLC ART scan.
+  //
+  // TODO(thompsonbry) Eventually this will need to use the stack to
+  // reconstruct the key from the path from the root to this leaf.
+  // Right now it is relying on the fact that simple fixed width keys
+  // are stored directly in the leaves.
+  if (!valid()) return {};  // not positioned on anything.
+  const auto& e = stack_.top();
+  const auto& node = e.node;
+  UNODB_DETAIL_ASSERT(node.type() == node_type::LEAF);     // On a leaf.
+  const auto* const aleaf{node.ptr<detail::olc_leaf*>()};  // current leaf.
+  key_ =
+      aleaf->get_key().decode();  // decode the key into the iterator's buffer
+  return key_;                    // return pointer to the internal key buffer.
+}
+
+inline std::optional<const value_view> olc_db::iterator::get_val()
+    const noexcept {
+  // Note: If the iterator is on a leaf, we return the value for
+  // that leaf regardless of whether the leaf has been deleted.
+  // This is part of the design semantics for the OLC ART scan.
+  //
+  if (!valid()) return {};  // not positioned on anything.
+  const auto& e = stack_.top();
```

Comment:
Same comment as for the other flavor re. `get_child` helper

Reply:
Per above. I do not see it as worth it here.  get_key() needs to change and there is no presumption that the reuse will remain